### ISSUE TITLE
Enable Manual Backups of the Database

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#2307]
We've always focused on automated backups (see #2305) that run on a schedule.  This is absolutely crucial to ensure that we have relatively frequent backups and do not persist data locally.  However, we might also expose the backup functionality to the client, so that a user could manually trigger a backup.  This would allow users to dump the latest dataset as they need it, and even download it to their local machine.

This has a number of advantages: 
 1. If the manual backup works, users can keep an up to date backup even if the automatic backup fails.
 2. By attempting manual backups, a user might be able to debug any problems with the automatic backup. 
 3. The users at Vanga seem so enjoy downloading the latest backup to run local analysis on.  Maybe more users will want this functionality built into the app.

The complications:
 1. Backing up takes a while ... we'll need to have a system where the user can schedule a backup and then get regular progress updates about how the backup is doing.  The logic here will likely be pretty complex and difficult to test.

**Comment by DedrickEnc** [comment]
@jniles 

Good Idea, it seems like this functionality needs some prerequisites such as user role functionality implementation. Now we have many user who have permission to see a specific set of data, implementing this functionality without caring about user role will give them all possibility to backup data in theory.

**Comment by jniles** [comment]
In my view, we can get away with this before implementing user roles by just putting it on a super user page, like the Enterprise Management page.  Do you think that would work for now?

**Comment by jeremielodi** [comment]
@jniles, as the role management module has been implemented, can we solve this issue?

**Comment by jniles** [comment]
Sure!  If you want to implement a solution go ahead!  Note the backups already exist in [server/lib/backup](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/lib/backup.js), so it just needs to be the UI side of this worked out.
